WikiClientLibrary/Generators/RandomPageGenerator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using WikiClientLibrary.Client;
+using WikiClientLibrary.Generators;
+using WikiClientLibrary.Generators.Primitive;
+using WikiClientLibrary.Pages;
+using WikiClientLibrary.Sites;
+
+namespace WikiClientLibrary.Generators
+{
+    /// <summary>
+    /// Gets random pages in a specific namespace.
+    /// </summary>
+    class RandomPageGenerator : WikiPageGenerator
+    {
+        /// <inheritdoc />
+        public RandomPageGenerator(WikiSite site) : base(site)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override WikiPageStub ItemFromJson(JToken json)
+        {
+            return new WikiPageStub((int)json["id"], (string)json["title"], (int)json["ns"]);
```

Comment:
You may use `MediaWikiHelper.PageStubFromJson`, which already properly handles the case.

Reply:
This method was deliberately overridden. Most other other queries return with a `pageid` attribute, 
```json
{
    "batchcomplete": "",
    "continue": {
        "apcontinue": "!!",
        "continue": "-||"
    },
    "query": {
        "allpages": [
            {
                "pageid": 5878274,
                "ns": 0,
                "title": "!"
            }
        ]
    }
}
```
but `random` returns an `id` attribute.
```json
{
    "batchcomplete": "",
    "continue": {
        "rncontinue": "0.106214315216|0.106214536074|22910380|0",
        "continue": "-||"
    },
    "query": {
        "random": [
            {
                "id": 56451309,
                "ns": 3,
                "title": "User talk:Ctextor at uofu"
            }
        ]
    }
}
```

Reply:
Confirmed. That's just wierd… Anyway then let it be.